core/Pos/Core/Fee.hs

```diff
@@ -0,0 +1,44 @@
+module Pos.Core.Fee
+       ( Coeff(..)
+       , TxSizeLinear(..)
+       , TxFeePolicy(..)
+       ) where
+
+import           Universum
+
+import           Data.Fixed          (Fixed (..), Nano, showFixed)
+import qualified Data.Text.Buildable as Buildable
+import           Formatting          (bprint, build, shown, (%))
+
+-- | A fractional coefficient of fixed precision.
+newtype Coeff = Coeff Nano
+    deriving (Eq, Show, Generic, NFData)
+
+instance Buildable Coeff where
+    build (Coeff x) = fromString (showFixed True x)
+
+-- | A linear equation on the transaction size. Represents the @\s -> a + b*s@
+-- function where @s@ is the transaction size, @a@ and @b@ are constant
+-- coefficients.
+data TxSizeLinear = TxSizeLinear !Coeff !Coeff
```

Comment:
Actually now I think that one value should be `CoinPortion` and another one should be `Coin` (when I wrote `Nano` in slack I didn't think at all).